src/socket_poller.cpp

```diff
@@ -401,7 +401,135 @@ void zmq::socket_poller_t::rebuild ()
     need_rebuild = false;
 }
 
-int zmq::socket_poller_t::wait (zmq::socket_poller_t::event_t *events_, int n_events_, long timeout_)
+void zmq::socket_poller_t::zero_trail_events (
```

Comment:
What about adding ```static inline``` descriptors? As @somdoron mentioned these functions get called a lot, so a hint to the compilers can't hurt.